vast/inline.go

```diff
@@ -44,15 +44,29 @@ func (inline *InLine) Validate() error {
 		}
 	}
 
-	for _, impression := range inline.Impressions {
-		if err := impression.Validate(); err != nil {
+	if len(inline.Impressions) == 0 {
+		errors = append(errors, ErrInlineMissImpressions)
+	}
+
+	// We don't want to over validate, as long as one impression contains a valid tracker
+	// we accept it.
+	var impressionErr []error
+	for i := range inline.Impressions {
+		if err := inline.Impressions[i].Validate(); err != nil {
 			ve, ok := err.(ValidationError)
 			if ok {
-				errors = append(errors, ve.Errs...)
+				impressionErr = append(impressionErr, ve.Errs...)
 			}
+		} else {
+			impressionErr = nil
```

Comment:
Isn't this huge loss of information? We lose the vases where errors take place. I would suggest checking for the error type in some way in Jaeger instead of making the library depend on it.

Reply:
we only log errors when the validation fails, but in this case validation "passes" so we don't log an error. It would be more complex if we wanted to capture something like *warnings* which don't actually invalidate the VAST.